src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -170,9 +172,58 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Length;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                if (count == 0) return (IPartition<TResult>)Clone();
+                if (count >= _source.Length) return new EmptyPartition<TResult>();
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return count >= _source.Length ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
+
+            public bool TryGetElementAt(int index, out TResult element)
+            {
```

Comment:
It could be faster to use    TResult TryGetElementAt(int index, out bool success).

If TResult happens to be a reference type, writing to out param would need to go through a checked GC fence. Returning TResult  will not need that since it is definitely on the stack. 
Fences are not very expensive, but if taken per-element, they can add up.